### PR TITLE
Paint the chrome before rastering it

### PR DESCRIPTION
**Pull request by pavpanchekha**
This PR changes how the browser chrome is drawn. Currently, the browser chrome is rastered directly, without an intermediate display list. This PR makes it create an intermediate display list first, made up of normal display commands, and then separately raster it.

The benefit of this is that the transition to Skia is much simpler, because we don't need separate `draw_line`, `draw_text`, and `draw_rect` commands, plus their (odd) calling conventions. (I think they also had a bug?) This makes the drawing command abstraction clearer and tighter.

The cost—the reason why we didn't do it earlier—is that we now need to define `DrawLine` and `DrawOutline` earlier. But we ended up needing them anyway, so I think it's not a big cost. I also changed the back button to draw a `<` character instead of drawing a polyline of that shape.

### CODE REVIEW

**Comment by pavpanchekha** [comment]
It's done. (Working on final tests now.) In the process of doing this I made a few bigger changes:

- I ended up changing up the order of chrome drawing in Chapter 7 quite a bit, since we now need new drawing commands and also a new `paint_chrome` method. But I think the new text is still good.
- I changed the constructor for `DrawOutline` take x/y/x/y instead of a `Rect`, so that it is the same as `DrawRect`. It's a bit weird when we switch to Skia later but it seemed better not to mess with it.
- In Chapter 11, the section on using Skia as the canvas got a lot shorter, so I moved in the related text/font stuff into that section. That left very little in the next section, but I guess for now I kept it as a breather. I also switched around "go further" blocks to better fit the content.
- In Chapter 14, doing all-black outlines first and then adding colored outlines is a bit less impactful now, since the underlying `DrawOutline` supports them from the start (which I did so it is similar to the other `Draw` commands), but I think it basically still works.

File: book/chrome.md [hunk 1]
```diff
@@ -622,29 +622,120 @@ happen in the `Browser` class.
...
+class Tab:
+    def scrolldown(self):
+        max_y = self.document.height - (HEIGHT - CHROME_PX)
+        self.scroll = min(self.scroll + SCROLL_STEP, max_y)
+```
+
+To better separate the chrome from the page, let's also add a border:
+
+``` {.python}
+CHROME_PX = 100
```
Comment (chrishtr) [inline book/chrome.md:688]: remove this as redundant given line 649 above

File: book/visual-effects.md [hunk 2]
```diff
@@ -289,218 +289,92 @@ use; modern browsers support [quite a lot][css-colors].
...
 
-Next, the plus button for adding a new tab:[^move-plus]
-
-[^move-plus]: I also changed the *y* position of the plus sign.
-    Skia draws fonts a bit differently from Tkinter, and the new *y*
-    position keeps the plus centered in the box. Feel free to adjust
-    the positions of the UI elements to make everything look good on
-    your system.
+To draw just the outline, set the `Style` parameter of the `Paint` to
+`Stroke_Style`. Here "stroke" is a standard term refering to drawing
```
Comment (chrishtr) [inline book/visual-effects.md:337]:
```suggestion
`Stroke_Style`. Here "stroke" is a standard term referring to drawing
```

**Review by chrishtr — COMMENTED** [review]
I assume you also tested all the chapters and they drew correctly? Do the widgets look ok?

**Comment by pavpanchekha** [comment]
The chapters do all draw correctly.

**Comment by pavpanchekha** [comment]
Ok, I checked the widgets and found two bugs. (Also, the fonts in the widgets are misaligned—I think that was already there—and there's some weird rendering behavior in lab11 which I also think was already there.

**Pull request merged**